lib/fog/vsphere/models/compute/server.rb

```diff
@@ -157,7 +157,11 @@ def clone(options = {})
 
           # Give our path to the request
           req_options['template_path'] = "#{relative_path}/#{name}"
-          req_options['datacenter'] = datacenter.to_s
+          if req_options['datacenter']
+            req_options['template_datacenter'] = datacenter.to_s
+          else
+            req_options['datacenter'] = datacenter.to_s
+          end
```

Comment:
I think, we always want to set the `template_datacenter`, right? :thinking:
What about this:
```suggestion
          req_options['template_datacenter'] = datacenter.to_s
          req_options['datacenter'] ||= datacenter.to_s
```

Reply:
I can change that, I thought it wouldn't be necessary because vm_clone uses options['template_datacenter'] || options['datacenter'], so template_datacenter was needed only if datacenter may be different, but it may be better to set template_datacenter explicitly